dali/dfuplus/dfuplus.cpp

```diff
@@ -1699,20 +1701,36 @@ int CDfuPlusHelper::erasehistory()
 
         int ofile = open(dstxml, _O_WRONLY | _O_CREAT | _O_TRUNC, _S_IREAD | _S_IWRITE);
         if(ofile == -1)
-            throw MakeStringException(-1, "can't open file %s\n", dstxml);
+            throw MakeStringException(-1, "can't open file %s, erase history cancelled.\n", dstxml);
 
         ssize_t written = write(ofile, xmlDump.str(), xmlDump.length());
         if (written < 0)
-            throw MakeStringException(-1, "can't write to file %s\n", dstxml);
+            throw MakeStringException(-1, "can't write to file %s, erase history cancelled.\n", dstxml);
 
         if (written != xmlDump.length())
-            throw MakeStringException(-1, "truncated write to file %s\n", dstxml);
+            throw MakeStringException(-1, "truncated write to file %s, erase history cancelled.\n", dstxml);
 
         close(ofile);
 
         info("History written into %s.\n",dstxml);
     }
 
+    Owned<IClientEraseHistoryRequest> req = dfuclient->createEraseHistoryRequest();
```

Comment:
Yes, that is true; there is a window between to list and erase operations, but what can happen?
- The logical file erased -> It is OK, the onEraseHitory() throws a file not found exception
- The logical file renamed -> same as before
- The logical file copied -> the copy has an original history + a new record about the copy. I don't think it can be a problem.
- A spray initiated to overwrite the logical file -> Perhaps in this case if the spray finished before the onEreaseHistory() call the first, one and only history record about the spray/import will be deleted. However the original aim behind the erase history feature was to remove sensitive information when the file migrated to other system, therefore I hardly believe those two operation would be executed in same time.

I can't see any other problematic case. 
@jakesmith @richardkchapman any comment?